core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -81,7 +81,7 @@
 
   /** Executor for application processing. */
   private Executor executor;
-  @GuardedBy("lock") private boolean usingSharedExecutor;
+  private boolean usingSharedExecutor;
```

Comment:
Adding `@GuardedBy` to `executor` seems an overkill and not something we'd want to do. `executor` is used on a fast-path that currently is not using `lock`. Simply put, neither variable is or should be guarded by the lock.

It actually looks like the executor can be used after releasing (I just created #2034) which means we can't set it to null until that is fixed first (and even then I'm not sure if it is worth-while).